pandas/tests/io/parser/test_common.py

```diff
@@ -1927,3 +1927,17 @@ def test_read_table_deprecated(all_parsers):
                                     check_stacklevel=False):
         result = parser.read_table(StringIO(data))
         tm.assert_frame_equal(result, expected)
+
+
+def test_first_row_bom_python(all_parsers):
+    parser = all_parsers
+    data = """
+    \ufeff"Head1"	"Head2"	"Head3"
+    """
+
+    assert parser.read_csv(StringIO(data),
```

Comment:
Can you construct the actual DataFrame that should be represented and do `tm.assert_frame_equal(result, expected)`? Should see that idiom in other tests in this module if you need some examples to look at

Reply:
Yes, I saw how it was done above. But the problem was in the correct calculation of columns, DataFrame was built correctly, if the check will be in the form of `tm.assert_frame_equal(result.shape, expected.shape)` will be fine?

Reply:
If I did this, would it be okay?

```
    assert result_c.shape == result_python.shape
    tm.assert_frame_equal(result_c, expected)
    tm.assert_frame_equal(result_python, expected)
```

Reply:
```python
result = parser.read_csv(StringIO(data), delimiter='\t')
expected = DataFrame(columns=["Head1", "Head2", "Head3"])
tm.assert_frame_equal(result, expected)
```

I think should work. Note that the parametrization for `all_parsers` makes it so you don't need to specify the engine as an argument